riocli/device/execute.py

```diff
@@ -101,17 +101,43 @@ def execute_command(
 
     device_guids = [d.uuid for d in devices if d.status == "ONLINE"]
     device_dict = {d.uuid: d.name for d in devices}
-    cmd = join(tuple(["bash", "-c"]) + command)
-    try:
-        result = client.execute_command(
-            device_ids=device_guids,
-            command=Command(cmd, shell=shell, bg=bg, run_async=run_async, runas=user, timeout=timeout),
-            timeout=timeout,
+    cmd = join(("bash", "-c", *command))
+    max_retries = 3
+
+    for _ in range(max_retries):
```

Comment:
question: This loop will end up re-executing the commands on the devices where the previous command didn't return results. This is not the desired behaviour.

Some devices might be late in responding to Saltstack. We should just poll for the results not execute_command again.